app/src/main/java/io/neurolab/main/output/visual/NeuroGameRenderer.java

```diff
@@ -0,0 +1,258 @@
+import com.jogamp.opengl.glu.GLU;
+import com.jogamp.opengl.util.gl2.GLUT;
+import com.jogamp.opengl.util.texture.Texture;
+
+public class NeuroGameRenderer {
+
+    public static Random rnd = new Random();
+    public static int numHorizontalQuads = 30;
+    public static int numDepthQuads = 50;
+    public static float[][] landscape;
+    public static int pointer = 0;
+    private static float[] fogcol = {0, .5f, .5f, 1};
+    private static float correctedFeedback = 0f;
+    private static float slowFeedback = 0f;
+    private static Random random = new Random(); // Seed to 0 for testing
+    public GLProfile glprofile;
+    int currentTheme = 0;
+    private int defaultTexture = 0;
+    private float currentFeedback = 0;
+    private GLUT glut;
+    private int width = 2;
+    private int height = 2;
+    private Texture texYantra;
+    private int shortSide;
+    private int shortSideHalf;
+    private int offsetX;
+    private int offsetY;
+    private Texture[] textures;
+    private int currentForestIn = 2;
+    private int currentForestOut = 1;
+    private Texture noiseTexture = null;
+    private float angle = 0f;
+    private float slowAngle = 0f;
+    private String[] themes = {"nforest", "universe"};
+    private float oldFeedback = 0f;
+
+    public NeuroGameRenderer(GLProfile glprofile2) {
+        this.glprofile = glprofile2;
+    }
+
+    public void nextTheme() {
+        currentTheme++;
+        if (currentTheme + 1 > themes.length)
+            currentTheme = 0;
+    }
+
+    public void setCurrentFeedback(float currentFeedback) {
+        this.currentFeedback = currentFeedback;
+    }
+
+    public void setup(GL2 gl2, int width, int height) {
+        if (landscape == null) {
+            landscape = new float[numDepthQuads][numHorizontalQuads];
+            for (int i = 0; i < numDepthQuads; i++) {
+                for (int j = 0; j < numHorizontalQuads; j++) {
+                    landscape[i][j] = rnd.nextFloat();
+                }
+            }
+        }
+
+        angle = 0f;
+        glut = new GLUT();
+        GLU glu = new GLU();
+
+        float aspect = (float) width / height;
+
+        // Set the view port (display area) to cover the entire window
+        gl2.glViewport(0, 0, width, height);
+
+        // Setup perspective projection, with aspect ratio matches viewport
+        gl2.glMatrixMode(GL2.GL_PROJECTION); // choose projection matrix
+        gl2.glLoadIdentity(); // reset projection matrix
+        glu.gluPerspective(75.0, aspect, 1, 1400.0); // fovy, aspect, zNear, zFar
+
+        // Enable the model-view transform
+        gl2.glMatrixMode(GL2.GL_MODELVIEW);
+        gl2.glLoadIdentity(); // reset
+
+        this.width = width;
+        this.height = height;
+
+        if (texYantra == null) {
+            texYantra = generateWhiteNoise(gl2, 800, 800);
+        }
+
+        if (width < height) {
+            shortSide = width;
+            offsetX = 0;
+            offsetY = height / 2;
```

Comment:
@suresh-gandhi I think these are again UI related. Should be done when we code the Fragments.